app/webpacker/scripts/form-check-leave.js

```diff
@@ -0,0 +1,30 @@
+function FormCheckLeave($module) {
+  this.$module = $module;
+}
+
+FormCheckLeave.prototype.init = function() {
+  const $form = this.$module;
+
+  if ($form) {
+    const $originalFormContent = encodeURIComponent(
+      Array.from(new FormData($form))
+    );
+
+    window.onbeforeunload = function(event) {
+      const $updatedFormContent = encodeURIComponent(
+        Array.from(new FormData($form))
```

Comment:
I think `Array.from` will break on IE11 (which is about 17% of users):
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from#Browser_compatibility

Reply:
What's your recommended fix? It was this vs a giant for loop which looked fragile.

For this type of function I'd recommend jQuery `serialize` but that's not my call to make.

Reply:
I think something like this would be fine; https://gomakethings.com/how-to-serialize-form-data-with-vanilla-js/. The jQuery method is convenient but we don't want to add jQuery.

Reply:
Requires all fields in the array to have a `name` attribute. They all should have so reusability shouldn't be an issue but just worry it's easy to break. I'll have a play around.

Reply:
We could use an Array.from polyfill? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from#Polyfill

Reply:
Got what Dan sent working but would just like to trial what I've pushed up. The use of `change` in the event listener is a cleaner solution that's worked for me.